packages/gatsby-cli/src/reporter/index.ts

```diff
@@ -40,76 +58,103 @@ if (process.env.GATSBY_LOGGER.includes(`json`)) {
   require(`./loggers/ink`)
 }
 
-const util = require(`util`)
-const { stripIndent } = require(`common-tags`)
-const chalk = require(`chalk`)
-const { trackError } = require(`gatsby-telemetry`)
-const tracer = require(`opentracing`).globalTracer()
-
-const { getErrorFormatter } = require(`./errors`)
-const { getStore } = require(`./redux`)
-import constructError from "../structured-errors/construct-error"
-
 const errorFormatter = getErrorFormatter()
 
-import type { ActivityTracker, ActivityArgs, Reporter } from "./types"
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
+export interface ActivityTracker {
+  start(): void
+  end(): void
+  span: Span
+  setStatus(status: string): void
+  panic: (errorMeta: ErrorMeta, error?: Error) => never
+  panicOnBuild: (errorMeta: ErrorMeta, error?: Error) => Error | undefined
+}
 
-const addMessage = level => text => reporterActions.createLog({ level, text })
+export type ProgressActivityTracker = Omit<ActivityTracker, "end"> & {
```

Comment:
I think this qualifies as "complicated types", I honestly would prefer to just see the fields copy pasted.

Reply:
Yeah, I think you're right. This was taken directly from the existing .d.ts types